src/Metrics.php

```diff
@@ -42,16 +42,23 @@ class Metrics
     /** @var string */
     private $package_version;
 
-    public function __construct(string $api_key, Closure $group, array $options = [])
+    /**
+     * @param string $api_key
+     * @param class-string $group_handler
+     * @param array $options
+     */
+    public function __construct(string $api_key, string $group_handler, array $options = [])
     {
         $this->api_key = $api_key;
-        $this->group = $group;
+        $this->group_handler = $group_handler;
         $this->development_mode = array_key_exists('development_mode', $options)
             ? (bool)$options['development_mode']
             : false;
 
-        $this->blacklist = array_key_exists('blacklist', $options) ? $options['blacklist'] : [];
-        $this->whitelist = array_key_exists('whitelist', $options) ? $options['whitelist'] : [];
+        $this->blacklist = array_key_exists('blacklist', $options) && is_array($options['blacklist'])
```

Comment:
`blacklist` and `whitelist` are already initialized as arrays. You might want to remove the ternary for:

```php
if(isset($options['blacklist'] && is_array($options['blacklist'])) {
    $this->blacklist = $options['blacklist'];
}
```